sound/soc/sof/ipc4-topology.c

```diff
@@ -1957,6 +1957,46 @@ static void sof_ipc4_put_queue_id(struct snd_sof_widget *swidget, int queue_id,
 	ida_free(queue_ida, queue_id);
 }
 
+static int sof_ipc4_set_copier_sink_format(struct snd_sof_dev *sdev,
+					   struct snd_sof_widget *swidget, int sink_id)
+{
+	struct sof_ipc4_copier *ipc4_copier = swidget->private;
+	struct sof_ipc4_copier_config_set_sink_format format;
+	struct sof_ipc4_fw_module *fw_module;
+	struct sof_ipc4_msg msg = {{ 0 }};
+	u32 header, extension;
+
+	dev_dbg(sdev->dev, "%s set copier sink %d format\n", swidget->widget->name, sink_id);
+
+	fw_module = swidget->module_info;
+
+	format.sink_id = sink_id;
+	memcpy(&format.source_fmt, &ipc4_copier->data.base_config.audio_fmt,
+	       sizeof(format.source_fmt));
+	memcpy(&format.sink_fmt, &ipc4_copier->data.out_format,
+	       sizeof(format.sink_fmt));
+	msg.data_size = sizeof(format);
+	msg.data_ptr = &format;
+
+	header = fw_module->man4_module_entry.id;
+	header |= SOF_IPC4_MOD_INSTANCE(swidget->instance_id);
+	header |= SOF_IPC4_MSG_TYPE_SET(SOF_IPC4_MOD_LARGE_CONFIG_SET);
+	header |= SOF_IPC4_MSG_DIR(SOF_IPC4_MSG_REQUEST);
+	header |= SOF_IPC4_MSG_TARGET(SOF_IPC4_MODULE_MSG);
+
+	extension = 0;
+	extension |= SOF_IPC4_MOD_EXT_MSG_SIZE(msg.data_size);
```

Comment:
why not remove the previous line and start with extension =...

Reply:
ack